persistence/schema/gov.go

```diff
@@ -3,440 +3,210 @@ package schema
-		message_unstake_service_node_fee_owner TEXT NOT NULL,
-		message_pause_service_node_fee_owner TEXT NOT NULL,
-		message_unpause_service_node_fee_owner TEXT NOT NULL,
-		message_change_parameter_fee_owner TEXT NOT NULL,
-		height BIGINT NOT NULL
-	)`
+	ParamsTableName   = "params"
+	ParamsTableSchema = `(
+		name VARCHAR(64) NOT NULL,
+		height BIGINT NOT NULL,
+		type val_type NOT NULL,
+		value TEXT NOT NULL,
+		PRIMARY KEY(name, height)
+		)`
+
+	FlagsTableName   = "flags"
+	FlagsTableSchema = `(
+		name VARCHAR(64) NOT NULL,
+		height BIGINT NOT NULL,
+		type val_type NOT NULL,
+		value TEXT NOT NULL,
+		enabled BOOLEAN NOT NULL,
+		PRIMARY KEY(name, height)
+		)`
 )
 
 var (
-	SQLColumnNames = []string{
-		"blocks_per_session",
-
-		"app_minimum_stake",
-		"app_max_chains",
-		"app_baseline_stake_rate",
-		"app_staking_adjustment",
-		"app_unstaking_blocks",
-		"app_minimum_pause_blocks",
-		"app_max_pause_blocks",
-
-		"service_node_minimum_stake",
-		"service_node_max_chains",
-		"service_node_unstaking_blocks",
-		"service_node_minimum_pause_blocks",
-		"service_node_max_pause_blocks",
-		"service_nodes_per_session",
-
-		"fisherman_minimum_stake",
-		"fisherman_max_chains",
-		"fisherman_unstaking_blocks",
-		"fisherman_minimum_pause_blocks",
-		"fisherman_max_pause_blocks",
-
-		"validator_minimum_stake",
-		"validator_unstaking_blocks",
-		"validator_minimum_pause_blocks",
-		"validator_max_pause_blocks",
-		"validator_maximum_missed_blocks",
-
-		"validator_max_evidence_age_in_blocks",
-		"proposer_percentage_of_fees",
-		"missed_blocks_burn_percentage",
-		"double_sign_burn_percentage",
+	govParamMetadataMap  map[string]govParamMetadata
+	govParamMetadataKeys []string
+
+	ValTypeEnumTypes = fmt.Sprintf(`(
+		'%s',
+		'%s',
+		'%s'
+		)`,
+		ValTypeString,
+		ValTypeBigInt,
+		ValTypeSmallInt,
+	)
+)
 
-		"message_double_sign_fee",
-		"message_send_fee",
-		"message_stake_fisherman_fee",
-		"message_edit_stake_fisherman_fee",
-		"message_unstake_fisherman_fee",
-		"message_pause_fisherman_fee",
-		"message_unpause_fisherman_fee",
-		"message_fisherman_pause_service_node_fee",
-		"message_test_score_fee",
-		"message_prove_test_score_fee",
-		"message_stake_app_fee",
-		"message_edit_stake_app_fee",
-		"message_unstake_app_fee",
-		"message_pause_app_fee",
-		"message_unpause_app_fee",
-		"message_stake_validator_fee",
-		"message_edit_stake_validator_fee",
-		"message_unstake_validator_fee",
-		"message_pause_validator_fee",
-		"message_unpause_validator_fee",
-		"message_stake_service_node_fee",
-		"message_edit_stake_service_node_fee",
-		"message_unstake_service_node_fee",
-		"message_pause_service_node_fee",
-		"message_unpause_service_node_fee",
-		"message_change_parameter_fee",
+// InsertParams generates the SQL INSERT statement given a *genesis.Params
+//
+// It leverages metadata in the form of struct tags (see `parseGovProto` for more information).
+//
+// WARNING: reflections in prod
```

Comment:
**Optional Suggestion** We could change the structure of the genesis.Params to a `[]Param` then we could loop over the param structure in order to remove reflection from prod.

I think the current approach is fine right now